src/radical/pilot/compute_unit_description.py

```diff
@@ -418,6 +420,20 @@ def __init__(self, from_dict=None):
     # --------------------------------------------------------------------------
     #
     def _verify(self):
+        
+        if callable(self.get('executable')):
+
+           exe = self.get('executable')
+           block = inspect.getsource(exe)
+
+           cu_exe_dict   = {'_cud_name':exe.__name__,
+                            '_cud_code':block,
+                            '_cud_args':'',
+                            '_cud_kwargs':''}
+
+
+           func_obj = codecs.encode(pickle.dumps(cu_exe_dict), "base64").decode()
+           self['executable'] = func_obj
```

Comment:
I think we should consider `verify` optional, and to not be used in non-debug runs.  The purpose is to *verify* data, not to *transform* data.    I understand that we don't have a good place to put that code.  Maybe something like this:
```py
descr = rp.ComputeUnitDescription()
descr.function = rp.prep_function(mathma)
desc, args = [7, 8],
descr.kwargs = {'verbose'=True}
```
?  Lets discuss.  @mtitov @mturilli @iparask : any opinion?

Reply:
One more argument for keeping verify optional: it is costly.  It is good to have, but if we **have** to use it for millions of tasks, we give away the performance advantage we gained with our `ru.Description` implementation...